doc/C_STYLE_GUIDE.rst.txt

```diff
@@ -177,14 +171,21 @@ Code lay-out
          * explanatory comment.
          */
 
+  or::
+
+        //
+        // This would be a long
+        // explanatory comment.
+        //
```

Comment:
C99 allows one line `//` comments I believe, not this. And given that we don't want to convert the current long comment style, it would probably be nicer to leave this out for consistency (although we can accept major code contributions that have it like this).

Reply:
> C99 allows one line // comments I believe, not this.

Each line is a single line comment, so C99 certainly allows that.  If NumPy prefers to disallow that style, that's a separate issue.  I don't see anything wrong with it, but I'll go with what y'all prefer.

I guess *mixing* different commenting styles within the same file looks funny and could be a distraction, but I think style guides generally recommend following the prevailing style of an existing file anyway.

Reply:
Yes, @charris may have an opinion. I would lean towards single line comments as well, just because that is what we have right now. (and frankly for a long block comment I probably even prefer the other style. I could even go with mostly recommending C++ style comments for `int i = 0  // counter` style in-line comments.

Does SciPy have a recommendation? Keeping in sync could be nice.

Reply:
no, SciPy simply follows NumPy in matters like these